src/private_key.rs

```diff
@@ -160,9 +160,9 @@ impl PrivateKey {
         let recovery_id = recovery_id.to_i32();
         assert!(recovery_id >= 0);
         let recovery_id = recovery_id as u32;
-        let v: Uint256 = (recovery_id + 27).into();
-        let r = Uint256::from_bytes_be(&compact[0..32]);
-        let s = Uint256::from_bytes_be(&compact[32..64]);
+        let v = Uint256::from_u32(recovery_id + 27);
+        let r = Uint256::from_bytes_be(&compact[0..32]).unwrap();
```

Comment:
is it safe to just unwrap here?

Reply:
It is safe because the `0..32` range is no more than 32 bytes. The indexing itself is what to be concerned about, but `compact` is 64 bytes so we can statically know it will not be out of bounds.